manifests/pnp4nagios/config.pp

```diff
@@ -0,0 +1,69 @@
+class icinga2::pnp4nagios::config inherits icinga2::pnp4nagios {
+
+  $monitoring_type = 'icinga'
+  $log_type = 'syslog'
+  $debug_lvl = 1
+
+  file {$htpasswd_config:
+    path   => "$httpd_path/$htpasswd_config",
+    ensure => $htpasswd_config_ensure,
+    owner  => $htpasswd_user,
+    group  => $htpasswd_group,
+    mode   => $htpasswd_config_mode,
+    source => "puppet:///modules/${module_name}/pnp/$htpasswd_config",
+    notify => Service['httpd'],
+  }
+
+  file { '/etc/nagios/passwd':
+    ensure  => file,
+    owner   => 'root',
+    group   => 'apache',
+    source  => "puppet:///modules/${module_name}/pnp/pnp_auth.txt",
```

Comment:
if you are creating /etc/nagios/passwd then your source filename should be passwd not pnp_auth.txt